coremltools/converters/mil/mil/ops/defs/complex_dialect_ops.py

```diff
@@ -732,13 +732,128 @@ class complex_shape(Operation):
     def type_inference(self):
         if not isinstance(self.x, ComplexVar):
             raise ValueError("x must be a ComplexVar.")
-        input_rank = self.x.real.rank
+        input_rank = self.x.rank
```

Comment:
If type_inference or value_inference is invoked when the graph is being constructed, x.real and x.imag may not be set since the complex lowering pass hasn't yet been invoked.